### DISCUSSION TITLE
Long Term Statistics

### DISCUSSION THREAD

**Discussion by andyblac** [BottlecapDave/HomeAssistant-OctopusEnergy#642] [General]
@BottlecapDave do you have any ideas why `sensor.octopus_energy_gas_xxxxx_xxxxxx_current_consumption`, would not show in the new History dashboard for LTS?, it does have the data according to Statistics in dev view

<img width="520" alt="Screenshot 2023-12-18 at 13 03 32" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/b3e6b20b-79a6-4de4-a869-332d615207d3">

but it not showing...
<img width="1555" alt="Screenshot 2023-12-18 at 13 02 44" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/ff4efd19-b936-4831-9644-0535b28fd903">

it is also the same with Electric.

The `sensor.octopus_energy_gas_xxxxxx_xxxxxx_current_accumulative_consumption` seems fine.
<img width="1555" alt="Screenshot 2023-12-18 at 13 13 46" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/df509e27-0d71-4f63-a6cc-7521cd9a6cba">

**Comment by andyblac** [comment]
```yaml
state_class: total
last_reset: "2023-12-18T13:53:04.226073+00:00"
unit_of_measurement: kWh
device_class: energy
icon: mdi:fire
friendly_name: Gas Current Consumption
last_evaluated: "2023-12-18T13:53:04.226073+00:00"
```
I see your using `state_class: total` for the sensor, should it not be `state_class: measurement` ??

> Measurement: The state represents a measurement in present time, not a historical aggregation such as statistics or a prediction of the future. Examples of what should be classified measurement are: current temperature, humidity or electric power.

> Total:  This state class should not be used for sensors where the absolute value is interesting instead of the accumulated growth or decline....

<img width="986" alt="Screenshot 2023-12-18 at 13 45 13" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/d44c96c5-3546-4595-890c-e67f6287bc5c">

**Reply by BottlecapDave** [reply]
Hey. I'm not sure why they're not coming through. The measurement type is not right as it can't be used in the energy dashboard and you can't have energy as the device class. It might be the last reset isn't working properly.

Tbh I've been thinking about deprecating this sensor as it existed before the accumulative sensor was a thing. But I'll take a look when I get a chance.

**Reply by andyblac** [reply]
but we dont use this sensor in Energy Dashboard, we use Current Accumulative for that ???

**Reply by andyblac** [reply]
> Tbh I've been thinking about deprecating this sensor as it existed before the accumulative sensor was a thing. But I'll take a look when I get a chance.

what, why this sensor is really handy to see current usage hour to hour for GAS.

**Reply by BottlecapDave** [reply]
The sensor was originally the only sensor available due to how current consumption data was retrieved. It was built to be used in the energy dashboard, and some people still use it for this purpose. Therefore if anything changes it needs to change for this purpose. The fact the energy dashboard displays the data correctly means the LTS data is there and at least in that scenario in a form HA is expecting.

As for hourly views, I believe (but will need testing) that you can use a utility meter with an hourly reset with the accumulative sensor as the source.

Basically the sensor is hang over from the before times and consuming a lot of my time trying to keep it working for different use cases which could be used for other features. It's just a consideration atm, but it's good to get people's use cases so I can have alternatives ready if the time comes.

**Comment by borpin** [comment]
> would not show in the new History dashboard for LTS?

I've seen an error at startup re the sensors not being added to LTS data.

**Reply by andyblac** [reply]
the data is there in the Database. so is being added to LTS.

**Comment by andyblac** [comment]
on looking closer the data is there, but it is not correct, it does not correspond to what is in the DB.
<img width="1551" alt="Screenshot 2023-12-18 at 14 01 06" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/d29a9291-d6da-46f0-bd9d-fce0ef6184de">

I think it is being averaged, I think this is to do with the state type of total, it is expecting to increase or decrease over time.
